solr/solrj/src/java/org/apache/solr/common/util/ExecutorUtil.java

```diff
@@ -196,7 +198,13 @@ public void execute(final Runnable command) {
 
       String ctxStr = contextString.toString().replace("/", "//");
       final String submitterContextStr = ctxStr.length() <= MAX_THREAD_NAME_LEN ? ctxStr : ctxStr.substring(0, MAX_THREAD_NAME_LEN);
-      final Exception submitterStackTrace = enableSubmitterStackTrace ? new Exception("Submitter stack trace") : null;
+      final Exception submitterStackTrace;
+      if (enableSubmitterStackTrace) {
+        Exception grandParentSubmitter = submitter.get();
+        submitterStackTrace = new Exception("Submitter stack trace", grandParentSubmitter);
```

Comment:
Could we have a comment (or rename) w.r.t. the `grandParent` bit here?

At a glance `parent` is intuitive with the `get/construct/set` pattern. On second glance it seems the current method executes the `submitter.set` call in a pool, making this method the "parent" relative to the pool execution and the `submitter.get` result the "parent" relative to this method but the "grand parent" relative to the method that will run in the pool. Is that it, more or less?

Reply:
I added a test for this to demonstrate what is happening, not sure if that made things more clear or added to the confusion. Can you take a look at it and then let me know what kind of comments you think are useful here?

Reply:
Hmm, yes, the tests help to demonstrate what is happening. But explaining it in a comment, well, that is trickier and so yes `grandParentSubmitter` is concise and comments as well might just add confusion.